File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1
    queue = []
    
    for i in s:
        if not queue:
            queue.append(i)
        else:
            if queue[-1] == i:
                queue.pop()
            else:
                queue.append(i)
    

    return 1 if not queue else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
class Stack {
    constructor() {
        this.value = []
        this.length = 0
    }

    push(value) {
        if (this.length === 0) {
            this.value.push(value)
            this.length++
        } else {
            
            if (this.value[this.length - 1] === value) {
                this.value.pop()
                this.length--
            } else {
                this.value.push(value)
                this.length++
            }
        }
    }
}

function solution(s){
    const stack = new Stack()
    s.split("").forEach(item => stack.push(item))

    return stack.length === 0 ? 1 : 0;
}